lib/binding/kvdb_interface.c

```diff
@@ -661,22 +672,24 @@ hse_kvdb_open(
     tstart = perfc_lat_start(&kvdb_pkvdbl_pc);
     perfc_inc(&kvdb_pc, PERFC_RA_KVDBOP_KVDB_OPEN);
 
-    err = argv_deserialize_to_kvdb_rparams(paramc, paramv, &params);
+    err = kvdb_rparams_from_paramv(&params, paramc, paramv);
     if (ev(err)) {
         log_errx("Failed to deserialize paramv for KVDB (%s) rparams", err, kvdb_home);
         return err;
     }
 
-    err = config_from_kvdb_conf(kvdb_home, &conf);
-    if (ev(err)) {
+    err = kvdb_home_get_config(kvdb_home, &config);
+    if (ev(err && merr_errno(err) != ENOENT)) {
         log_errx("Failed to read KVDB config file (%s/kvdb.conf)", err, kvdb_home);
         return err;
     }
 
-    err = config_deserialize_to_kvdb_rparams(conf, &params);
-    if (ev(err)) {
-        log_errx("Failed to deserialize config file for KVDB (%s) rparams", err, kvdb_home);
-        goto out;
+    if (config) {
+        err = kvdb_rparams_from_config(&params, config);
+        if (ev(err)) {
+            log_errx("Failed to deserialize config file for KVDB (%s) rparams", err, kvdb_home);
+            goto out;
+        }
```

Comment:
From a flow perspective I prefer function inputs to the left and outputs to the right...

Reply:
Which function are you referring to? `kvdb_rparams_from_config()`? It is an in/out parameter. It is updated from the values of the config